src/hotspot/share/runtime/synchronizer.cpp

```diff
@@ -118,312 +230,24 @@ int dtrace_waited_probe(ObjectMonitor* monitor, Handle obj, Thread* thr) {
 #define NINFLATIONLOCKS 256
 static volatile intptr_t gInflationLocks[NINFLATIONLOCKS];
 
-// global list of blocks of monitors
-PaddedObjectMonitor* ObjectSynchronizer::g_block_list = NULL;
+static MonitorList _in_use_list;
+// The ratio of the current _in_use_list count to the ceiling is used
+// to determine if we are above MonitorUsedDeflationThreshold and need
+// to do an async monitor deflation cycle. The ceiling is increased by
+// AvgMonitorsPerThreadEstimate when a thread is added to the system
+// and is decreased by AvgMonitorsPerThreadEstimate when a thread is
+// removed from the system.
+// Note: If the _in_use_list max exceeds the ceiling, then
+// monitors_used_above_threshold() will use the in_use_list max instead
+// of the thread count derived ceiling because we have used more
+// ObjectMonitors than the estimated average.
+//
+// Start the ceiling with the estimate for one thread:
+jint _in_use_list_ceiling = AvgMonitorsPerThreadEstimate;
```

Comment:
Why is this a jint when you use size_t for its accessor and all the other sizes that you compare with the ceiling are also size_t?
I'm not sure size_t is right to use in these cases (do we really expect different maximums on 32-bit versus 64-bit?) but it should be all or none IMO.

Reply:
Our int types are really confused.  AvgMonitorsPerThreadEstimate is defined as an intx which is intptr_t and the range of it is 0..max_jint which is 0 .. 0x7fffffff . jint is long on windows (the problematic type) and int on unix.  Since this is a new declaration, it probably should be something other than jint but what?
At any rate, it should be declared as 'static'.

Reply:
`_in_use_list_ceiling` is a jint **because** we've specified the range
as `0..max_jint` and I wanted some sanity to that variable's type.

If I change `_in_use_list_ceiling` to `size_t`, then I get a compile time
error probably because  `AvgMonitorsPerThreadEstimate` is an `intx`
which (I think) is my only choice for a command line option.

@fisk will have to chime in with the background on why he picked `size_t`.

Reply:
@coleenp - Nice catch on the missing 'static'.

Reply:
I typically use size_t for entities that can scale with the size of the machine's memory, so I don't have to think about whether there are enough bits. Could AvgMonitorsPerThreadEstimate be uintx instead of intx? And then maybe we don't need to declare a range, as the natural range of the uintx seems perfectly valid.

Reply:
I'm pretty sure I copied the decl for AvgMonitorsPerThreadEstimate
from some other already existing option. That's SOP for me anyway...
If we make any more changes here it will have to be in a follow up.

Reply:
Filed the following new RFE:
    JDK-8256307 cleanup AvgMonitorsPerThreadEstimate and _in_use_list_ceiling types
    https://bugs.openjdk.java.net/browse/JDK-8256307